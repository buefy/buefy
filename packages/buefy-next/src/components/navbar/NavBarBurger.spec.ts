import { beforeEach, describe, expect, it } from 'vitest'
import { shallowMount, mount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import BNavbarBurger from '@components/navbar/NavbarBurger.vue'

describe('BNavbarBurger', () => {
    let wrapper: VueWrapper<InstanceType<typeof BNavbarBurger>>
    beforeEach(() => {
        wrapper = shallowMount(BNavbarBurger)
    })

    it('is called', () => {
        expect(wrapper.vm).toBeTruthy()
        expect(wrapper.vm.$options.name).toBe('NavbarBurger')
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should be an anchor tag', () => {
        expect(wrapper.find('a.navbar-burger').exists()).toBeTruthy()
    })

    it('should have the is-active class when the prop isOpened is true', async () => {
        const burgerWrapper = mount(BNavbarBurger)
        await burgerWrapper.setProps({ isOpened: true })
        expect(burgerWrapper.classes('is-active')).toBe(true)
    })
})
