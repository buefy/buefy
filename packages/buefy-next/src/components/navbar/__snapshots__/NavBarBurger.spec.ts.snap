// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BNavbarBurger > render correctly 1`] = `"<a role=\\"button\\" class=\\"navbar-burger burger\\" aria-label=\\"menu\\" tabindex=\\"0\\"><span aria-hidden=\\"true\\"></span><span aria-hidden=\\"true\\"></span><span aria-hidden=\\"true\\"></span></a>"`;
